Rx.NET/Documentation/Rx-Roadmap-2023.md

```diff
@@ -0,0 +1,167 @@
+
+* VS complains that the project targets versions of .NET that are not installed, unless you install out-of-support .NET Core SDKs (going back to .NET Core 2.1)
+* If you do install the older .NET and .NET Core SDKs, Visual Studio will emit warnings telling you that the relevant frameworks are out of support and will not receive security updates
+* There are numerous problems with the UWP tests:
+  * The project enabling tests to execute on UWP (`Tests.System.Reactive.Uwp.DeviceRunner.csproj`) won't load due to incompatibilities with the version of `Coverlet.Collector` in use
+  * If you remove the reference to `Coverlet.Collector`, Visual Studio reports that the UWP test project's references to the `System.Reactive` and `Microsoft.Reactive.Testing` projects are not allowed (with no explanation as to why) although it appears to build anyway
+  * When you attempt to run the unit tests in Visual Studio's Test Explorer, none of the UWP tests run—they all just show a blue exclamation mark, with no explanation provided as to why they were not run; this is because Rx uses the [xUnit.net](https://xunit.net/) unit testing tools, and it appears that the [UWP support in these tools no longer works](https://github.com/xunit/devices.xunit/issues/171)—if you follow the [xUnit instructions for creating a brand new UWP project from scratch and adding a test](https://xunit.net/docs/getting-started/uwp/devices-runner) it does not work; it appears that the only update to the [xUnit UWP test runner](https://github.com/xunit/devices.xunit/commits/master) since November 2019 was a change to the logo; in fact the last release was in January 2019 (back when Visual Studio 2017 was still the latest version), with almost all subsequent updates to the repo being dependabot updates rather than new development
+* Some of the certificates used for building certain tests have expired
+* There are some warnings relating to C#'s nullable reference types feature
+* Newer analyzer rules cause a huge number of warnings and diagnostic messages to appear
+* Visual Studio modifies some Xamarin-related auto-generated files due to changes in the tool versions
+
+### Unnecessary bloat with certain build techniques
+
+If you use .NET to build a desktop application, and if your create a self-contained deployments, adding a reference to Rx can cause tens of megabytes of components to be added to your application unnecessarily. This has caused problems for the Avalonia team (see https://github.com/AvaloniaUI/Avalonia/issues/9549) with the result that they have removed Rx from their codebase (see https://github.com/AvaloniaUI/Avalonia/pull/9749).
+
+There are a few instances in which projects have effectively built in their own miniature versions of Rx because of the problems that can occur when using the real `System.Interactive`.
```

Comment:
I might swap this round so you pull the sentence about Avalonia from the previous paragraph as a specific example of people building their own "good bits" version of Rx.

Reply:
Done.